hathor/transaction/transaction_metadata.py

```diff
@@ -21,7 +22,73 @@
 if TYPE_CHECKING:
     from weakref import ReferenceType  # noqa: F401
 
-    from hathor.transaction import BaseTransaction  # noqa: F401
+    from hathor.transaction import BaseTransaction
+
+
+@unique
+class ValidationState(IntEnum):
+    """
+
+    Possible transitions:
+
+    - Initial
+        -> Basic: parents exist, graph information checks-out
+        -> Invalid: all information to reach `Basic` was available, but something doesn't check out
+        -> Checkpoint: is a block which hash matches a known checkpoint is a parent of a Checkpoint-valid tx
+    - Basic
+        -> Full: all parents reached `Full`, and validation+consensus ran successfully
+        -> Invalid: all information to reach `Full` was available, but something doesn't check out
+    - Checkpoint
+        -> Checkpoint-Full: when all the chain of parents and inputs up to the genesis exist in the database
+    - Full: final
+    - Checkpoint-Full: final
+    - Invalid: final
+
+    `BASIC` means only the validations that can run without access to the dependencies (parents+inputs, except for
+    blocks the block parent has to exist and be at least BASIC) have been run. For example, if it's `BASIC` the weight
+    of a tx has been validated and is correct, but it may be spending a tx that has already been spent, we will not run
+    this validation until _all_ the dependencies have reached `FULL` or any of them `INVALID` (which should
+    automatically invalidate this tx). In theory it should be possible to have even more granular validation (if one of
+    the inputs exists, validate that we can spend it), but the complexity for that is too high.
+
+    """
+    INITIAL = 0  # aka, not validated
+    BASIC = 1  # only graph info has been validated
+    CHECKPOINT = 2  # validation can be safely assumed because it traces up to a known checkpoint
```

Comment:
As long as the hash matches, right?

Reply:
Yes, I tried to keep the comment short, is so long already.

Reply:
Just double checking. Thanks!